cmd/vc-rest/startcmd/start.go

```diff
@@ -88,6 +86,34 @@ const (
 		" For CouchDB, include the username:password@ text if required." +
 		" Alternatively, this can be set with the following environment variable: " + databaseURLEnvKey
 
+	databasePrefixFlagName  = "database-prefix"
+	databasePrefixEnvKey    = "DATABASE_PREFIX"
+	databasePrefixFlagUsage = "An optional prefix to be used when creating and retrieving underlying databases." +
+		" Alternatively, this can be set with the following environment variable: " + databasePrefixEnvKey
+
+	// Linter gosec flags these as "potential hardcoded credentials". They are not, hence the nolint annotations.
+	kmsSecretsDatabaseTypeFlagName      = "kms-secrets-database-type" //nolint: gosec
+	kmsSecretsKeyDatabaseTypeEnvKey     = "KMSSECRETS_DATABASE_TYPE"  //nolint: gosec
+	kmsSecretsDatabaseTypeFlagShorthand = "k"
+	kmsSecretsDatabaseTypeFlagUsage     = "The type of database to use for storage of KMS secrets. " +
+		"Supported options: mem, couchdb. Alternatively, this can be set with the " +
+		"following environment variable: " + kmsSecretsKeyDatabaseTypeEnvKey
+
+	kmsSecretsDatabaseURLFlagName      = "kms-secrets-database-url" //nolint: gosec
+	kmsSecretsDatabaseURLEnvKey        = "KMSSECRETS_DATABASE_URL"  //nolint: gosec
+	kmsSecretsDatabaseURLFlagShorthand = "s"
+	kmsSecretsDatabaseURLFlagUsage     = "The URL of the database. Not needed if using memstore. For CouchDB, " +
+		"include the username:password@ text if required. It's recommended to not use the same database as the one " +
+		"set in the " + databaseURLFlagName + " flag (or the " + databaseURLEnvKey + " env var) since having access " +
+		"to the KMS secrets may allow the host of the provider to decrypt EDV encrypted documents. " +
+		"Alternatively, this can be set with the following environment variable: " + databaseURLEnvKey
+
+	kmsSecretsDatabasePrefixFlagName  = "kms-secrets-database-prefix" //nolint: gosec
+	kmsSecretsDatabasePrefixEnvKey    = "KMSSECRETS_DATABASE_PREFIX"  //nolint: gosec
+	kmsSecretsDatabasePrefixFlagUsage = "An optional prefix to be used when creating and retrieving " +
+		"the underlying KMS secrets database. Alternatively, this can be set with the following environment variable: " +
+		kmsSecretsDatabasePrefixEnvKey
+
```

Comment:
Since the whole point of an EDV is to prevent the host provider from being able to read your documents, I think the KMS secrets should be in a separate provider.  I don't prevent the user from setting both to the same provider, but if they do I log a warning (see the other comment I tagged you in).

Thoughts?

@troyronda @fqutishat @sandrask